core/arch/arm/kernel/boot.c

```diff
@@ -1385,6 +1390,16 @@ void init_tee_runtime(void)
 	 */
 	thread_init_core_local_pauth_keys();
 	thread_init_thread_pauth_keys();
+
+	/*
+	 * Reinitialize canaries around the stacks with crypto_rng_read().
+	 * For virtualization, we do not update canaries because otherwise we
+	 * will need a lock to avoid racing between thread_check_canaries
+	 * operate on nexus and thread_update_canaries operate on partition,
+	 * which will induce huge performance penalty.
```

Comment:
Really? It is hard to believe. My understanding was that is was more of a "TODO" thing, left for later...

Reply:
> 

Yes, we believe there must be ways to get it done, we just don't have a clever solution yet and rather to save it for another P-R.
We will rephrase the comment, such that it won't discourage other contributor from adding support of update canaries when `CFG_NS_VIRTUALIZATION` is enabled in the future.

Reply:
Very good, thanks